registry/auth.go

```diff
@@ -147,18 +152,58 @@ func decodeAuth(authStr string) (string, string, error) {
 
 // load up the auth config information and return values
 // FIXME: use the internal golang config parser
-func LoadConfig(rootPath string) (*ConfigFile, error) {
-	configFile := ConfigFile{Configs: make(map[string]AuthConfig), rootPath: rootPath}
-	confFile := path.Join(rootPath, CONFIGFILE)
+func LoadConfig(configDir string) (*ConfigFile, error) {
+	if configDir == "" {
+		configDir = filepath.Join(homedir.Get(), ".docker")
+	}
+
+	configFile := ConfigFile{
+		AuthConfigs: make(map[string]AuthConfig),
+		filename:    filepath.Join(configDir, CONFIGFILE),
+	}
+
+	// Try happy path first - latest config file
+	if _, err := os.Stat(configFile.filename); err == nil {
+		data, err := ioutil.ReadFile(configFile.filename)
+		if err != nil {
+			return &configFile, err
+		}
+
+		err = json.Unmarshal(data, &configFile)
+		if err != nil {
+			return &configFile, err
+		}
+
+		for addr, ac := range configFile.AuthConfigs {
+			ac.Username, ac.Password, err = decodeAuth(ac.Auth)
+			if err != nil {
+				return &configFile, err
```

Comment:
This was how the old code worked so I followed their lead.
If I had to guess I'd say it was because they didn't want to consider parsing the config file to be something worth stopping on.  We can change it but I'd prefer to do it in a separate PR because I have a sense it'll lead to a long bikeshedding discussion.  At least this PR doesn't change today's behavior.